src/Orleans/Runtime/AsynchQueueAgent.cs

```diff
@@ -30,7 +30,10 @@ public void QueueRequest(T request)
                 queueTracking.OnEnQueueRequest(1, requestQueue.Count, request);
             }
 #endif
-            requestQueue.Add(request);
+            if (requestQueue != null)
+            {
+                requestQueue.Add(request);
```

Comment:
For this one, it might be better to invert the logic and add a comment. Something like:
```C#
// Avoid adding to the queue during shutdown.
if (requestQueue == null) return;
```